packages/core/src/metrics/MetricsService.ts

```diff
@@ -47,4 +42,20 @@ export abstract class MetricsService {
         return this.publisher.publish(payload);
     }
 
+    /**
+     * Generates or gets mobile client id
+     */
+    public getClientId(): string {
+        let clientId = this.getStoredClientId();
+
+        if (!clientId) {
+            clientId = "TODO: generate a uuid";
+            this.storeClientId(clientId);
+        }
+
+        return clientId;
+    }
+
+    protected abstract getStoredClientId(): string;
```

Comment:
My vote will be `getClientId` and `saveClientId`. It could be stored or retriever from somewhere so generic names are better. Also add documentation.

Reply:
I would still leave 3 methods, 1 that generates a new ID and 2 abstract that get/save. Otherwise the contract between core and core-cordova is broken. How do we ensure it checks if clientId exists before creating a new one? We should leave loading and saving implementation for the specific skds, Cordova will use `window.localStorage` but React Native has to use native persistent storage ([AsyncStorage](https://facebook.github.io/react-native/docs/asyncstorage.html#docsNav)).

Reply:
I meant to rename them only.